docs/docs/async-dialogs.md

```diff
@@ -4,3 +4,628 @@ title: Async Dialogs
 ---
 
 [async-dialogs](unfinished-article)
+
+Typically *promises* are used in conjunction with asynchronous tasks such as a
+network request or a `setTimeout`; a lesser explored use is in dealing with user
+input. Since a program has to wait for a user to continue some actions it makes
+sense to consider it an asynchronous event.
+
+For comparison I'll start with an example of a *synchronous* user interaction
+using the `window.prompt` and then move to an *asynchronous* interaction my
```

Comment:
I don't think `the` is correct here - just "using `window.prompt`" sounds more correct.